x-pack/plugins/enterprise_search/public/applications/app_search/components/role_mappings/role_mappings_logic.test.ts

```diff
@@ -0,0 +1,437 @@
+          new Set(engines.map((e) => e.name))
+        );
+      });
+
+      it('sets default group with new role mapping', () => {
+        RoleMappingsLogic.actions.setRoleMappingData({
+          ...mappingServerProps,
+          roleMapping: undefined,
+        });
+
+        expect(RoleMappingsLogic.values.selectedEngines).toEqual(new Set());
+      });
+    });
+
+    it('handleRoleChange', () => {
+      RoleMappingsLogic.actions.handleRoleChange('dev');
+
+      expect(RoleMappingsLogic.values.roleType).toEqual('dev');
+    });
+
+    it('handleEngineSelectionChange', () => {
+      const engine = engines[0];
+      const otherEngine = engines[1];
+      RoleMappingsLogic.actions.setRoleMappingData({
+        ...mappingServerProps,
+        roleMapping: {
+          ...asRoleMapping,
+          engines: [engine, otherEngine],
+        },
+      });
+
+      RoleMappingsLogic.actions.handleEngineSelectionChange(otherEngine.name, true);
+      expect(RoleMappingsLogic.values.selectedEngines).toEqual(
+        new Set([engine.name, otherEngine.name])
+      );
+
+      RoleMappingsLogic.actions.handleEngineSelectionChange(otherEngine.name, false);
+      expect(RoleMappingsLogic.values.selectedEngines).toEqual(new Set([engine.name]));
+    });
+
+    it('handleAccessAllEnginesChange', () => {
+      RoleMappingsLogic.actions.handleAccessAllEnginesChange();
+
+      expect(RoleMappingsLogic.values.accessAllEngines).toEqual(false);
+    });
+
+    describe('handleAttributeSelectorChange', () => {
+      const elasticsearchRoles = ['foo', 'bar'];
+
+      it('sets values correctly', () => {
+        RoleMappingsLogic.actions.setRoleMappingData({
+          ...mappingServerProps,
+          elasticsearchRoles,
+        });
+        RoleMappingsLogic.actions.handleAttributeSelectorChange('role', elasticsearchRoles[0]);
+
+        expect(RoleMappingsLogic.values.attributeValue).toEqual(elasticsearchRoles[0]);
+        expect(RoleMappingsLogic.values.attributeName).toEqual('role');
+      });
+
+      it('correctly handles "role" fallback', () => {
+        RoleMappingsLogic.actions.handleAttributeSelectorChange('username', elasticsearchRoles[0]);
+
+        expect(RoleMappingsLogic.values.attributeValue).toEqual('');
+      });
+    });
+
+    it('handleAttributeValueChange', () => {
+      RoleMappingsLogic.actions.handleAttributeValueChange('changed_value');
+
+      expect(RoleMappingsLogic.values.attributeValue).toEqual('changed_value');
+    });
+
+    describe('handleAuthProviderChange', () => {
+      beforeEach(() => {
+        RoleMappingsLogic.actions.setRoleMappingData({
+          ...mappingServerProps,
+          roleMapping: {
+            ...asRoleMapping,
+            authProvider: ['foo'],
+          },
+        });
+      });
+      const providers = ['bar', 'baz'];
+      const providerWithAny = [ANY_AUTH_PROVIDER, providers[1]];
+      it('handles empty state', () => {
```

Comment:
optional nit, not sure if you care but mostly suggesting for consistency w/ the rest of the file

```suggestion
      const providerWithAny = [ANY_AUTH_PROVIDER, providers[1]];

      it('handles empty state', () => {
```